### PR TITLE
Adding Python Solution for Drought (1181)

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-1181.mdx [hunk 1]
```diff
@@ -101,5 +101,79 @@ int main() {
...
+		n_cows_sub = n_cows - 1
+
+	for i in range(2, n_cows_sub, 2):
+		corn[i] = (hunger_lvl[i] - corn[i - 1] - same_lvl)
+
+	works = check(same_lvl, corn, hunger_lvl)
+	return sum(corn) * 2 if works else -1
+
+
+def draught():
```
Comment (SansPapyrus683) [inline solutions/usaco-1181.mdx:159]: i'd recommend just putting this and `solve` into the actual body of the loop

File: solutions/usaco-1181.mdx [hunk 2]
```diff
@@ -101,5 +101,79 @@ int main() {
...
+	n_cows = int(input())
+	lvl = list(map(int, input().split()))
+
+	corns = [0] * (n_cows - 1)
+	if n_cows < 3:
+        	if len(set(lvl)) == 1:
+            		return 0
+        	else: 
+            		return -1
```
Comment (SansPapyrus683) [inline solutions/usaco-1181.mdx:168]:
```suggestion
		return 0 if len(set(lvl)) == 1 else -1
```

File: solutions/usaco-1181.mdx [hunk 3]
```diff
@@ -101,5 +101,79 @@ int main() {
 ```
         
 </CPPSection>
+
+<PySection>
+
+```python
+# Check if a proposed levelling works given some corn amounts.
+def check(same_lvl, corn, hunger_lvl):
```
Comment (SansPapyrus683) [inline solutions/usaco-1181.mdx:109]: type annotations

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1181.mdx (hunk 1 above)
Comment (jjliewie) [reply solutions/usaco-1181.mdx]: is it better now?

File: solutions/usaco-1181.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-1181.mdx]: done

File: solutions/usaco-1181.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-1181.mdx:108]: ?

Comment (SansPapyrus683) [reply solutions/usaco-1181.mdx:108]: https://docs.python.org/3/library/typing.html

Comment (jjliewie) [reply solutions/usaco-1181.mdx]: done

File: solutions/usaco-1181.mdx [hunk 4]
```diff
@@ -105,7 +105,7 @@ int main() {
 <PySection>
 
 ```python
-def check(same_lvl, corn, hunger_lvl):
+def check(same_lvl: int, corn: list, hunger_lvl: list) -> bool:
```
Comment (SansPapyrus683) [inline solutions/usaco-1181.mdx:108]:
```suggestion
def check(same_lvl: int, corn: [int], hunger_lvl: [int]) -> bool:
```

File: solutions/usaco-1181.mdx [hunk 5]
```diff
@@ -101,5 +101,65 @@ int main() {
...
+	hunger_lvl[0] -= corn[0]
+	hunger_lvl[-1] -= corn[-1]
+
+	for i in range(1, len(hunger_lvl)-1):
+		sum_of_corn = corn[i-1] + corn[i]
+		hunger_lvl[i]-= sum_of_corn
+
+	return len(set(hunger_lvl)) == 1 and same_lvl >= 0 and min(corn) >= 0
+
+def draught() -> int:
```
Comment (SansPapyrus683) [inline solutions/usaco-1181.mdx:119]: does this have to be in a function? ik python runs a bit faster when in a function

Comment (jjliewie) [reply solutions/usaco-1181.mdx]: changed

**Comment by SansPapyrus683** [comment]
did u test the updated sol?

**Comment by jjliewie** [comment]
> did u test the updated sol?

yeah sorry mybad forgot to add continue

**Comment by jjliewie** [comment]
it works now
Screen Shot 2022-05-10 at 1 16 10 PM

**Review by SansPapyrus683 — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
@ryanchou-dev

File: solutions/usaco-1181.mdx [hunk 6]
```diff
@@ -101,5 +101,63 @@ int main() {
...
+<PySection>
+
+```python
+def check(same_lvl: int, corn: [int], hunger_lvl: [int]) -> bool:
+	# check if proposed corn is achievable
+	hunger_lvl[0] -= corn[0]
+	hunger_lvl[-1] -= corn[-1]
+
+	for i in range(1, len(hunger_lvl)-1):
+		sum_of_corn = corn[i-1] + corn[i]
```
Comment (ryanchou-dev) [inline solutions/usaco-1181.mdx:114]: space out ops

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]

File: solutions/usaco-1181.mdx (hunk 6 above)
Comment (ryanchou-dev) [reply solutions/usaco-1181.mdx:114]: bump

File: solutions/usaco-1181.mdx [hunk 7]
```diff
@@ -155,7 +155,11 @@ for _ in range(int(input())):
 		corns[i] = (lvl[i] - corns[i - 1] - same_lvl)
 
 	works = check(same_lvl, corns, lvl)
-	print(sum(corns) * 2 if works else -1)
+	
+	if works:
+		print(sum(corns) * 2)
+	else:
+		print(-1)
```
Comment (SansPapyrus683) [inline solutions/usaco-1181.mdx:162]: why

Comment (jjliewie) [reply solutions/usaco-1181.mdx:162]: so it's easier to understand..

**Pull request merged**